frontend/libs/studio-components/src/components/StudioTreeView/StudioTreeViewItem/StudioTreeViewItem.test.tsx

```diff
@@ -151,6 +151,22 @@ describe('StudioTreeViewItem', () => {
     expect(getTreeItem({ name: label })).toBe(wrapper.querySelector('[role="treeitem"]')); // eslint-disable-line testing-library/no-node-access
   });
 
+  it('Opens the tree item when a child is selected', async () => {
```

Comment:
Is this test actually something like `Expands the item when selected and it can contain children`?

I dont really understand this premises of this test? Is it suppose to simulate adding a container item to a layout? if so maybe it is more correct simulating that the children prop is empty? 

But however, I see that the test is indeed testing the desired behavior for the PR so I guess it is good enough 😬

Reply:
> Is this test actually something like Expands the item when selected and it can contain children?

The test was made to test this case #12038, where my thought was that with two constraints:

* added nodes should be marked as selected
* selected nodes *or nodes with child nodes that are selected* should be expanded/opened

It would solve the issue without complicating the selection/expanding logic depending on where the node was added.

Basically it tests this:

* parent node <- expect this to be open
   * child node <- selected

Where before the parent would remain closed and the child node would not be visible even though it was selected.

I'm not sure if this is conveyed clearly enough in the code.

Reply:
Maybe we should have two tests? One for this case your are referring to and another more specific for the case where we add an empty parent that also should be expanded? 😊

Reply:
There already is, in a way. To my understanding, the `treeview` implementation delegates the adding/removing of elements completely to the implementation details. Which is why the relevant test for this user action is here: https://github.com/Altinn/altinn-studio/blob/14971fb1e3db4f4cbd0cf4f70ad7688e43fead2e/frontend/libs/studio-components/src/components/StudioTreeView/StudioTreeViewItem/StudioTreeViewItem.test.tsx#L91

Reply:
Hm, but does that test cover the case when the group component is added initially? Because if this test was there from before it should have failed for that case since these containers did not autoopen when added to the layout the first time 🤔

Reply:
No it does not, since adding and removing is handled by the implementor of `treeview` and not `treeview` itself. There could possibly be implemented some sort of signal sent from the implementation to treeview that something was added, but since i am not familiar with the intentions of the original design i did not want to write a patch possibly breaking the intended logic. The select state works as a catch-all for this. Which is tested in the implementation, like here:  https://github.com/Altinn/altinn-studio/blob/e8283562c11fb7a0e654e5fdb2dcecf3f2bd13ba/frontend/packages/schema-editor/src/components/SchemaTree/SchemaNode/SchemaNode.test.tsx#L167